### PR TITLE
[IMP] Migrate fields and records to a new module

### PR DESCRIPTION
**Pull request by mmequignon**
Some fields and models aren't ported from 13.0 to 14.0.
A new module introduced here https://github.com/OCA/community-data-files/pull/118
adds those fields and records, but we need a migration script as well.

### CODE REVIEW

File: l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    ],
+}
+ACCESS_NAMES_TO_MOVE = [
+    "access_storage_class",
+    "access_packaging_type",
+    "access_storage_temp",
+    "access_wgk_class",
+    "access_limited_amount",
+]
+MODULE_NAMES_TO_MOVE = (
```
Comment (sebalix) [inline l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py]:
```suggestion
MODELS_TO_MOVE = (
```

File: l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+def move_fields_to_new_module(cr):
+    for model, field_names in FIELD_NAMES_TO_MOVE.items():
+        openupgrade.update_module_moved_fields(
+            cr, model, field_names, PREVIOUS_MODULE_NAME, NEW_MODULE_NAME
+        )
+
+
+def move_records_to_new_module(cr):
+    cr.execute(QUERY_GET_NAMES, (PREVIOUS_MODULE_NAME, MODULE_NAMES_TO_MOVE))
```
Comment (sebalix) [inline l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py]:
```suggestion
    cr.execute(QUERY_GET_NAMES, (PREVIOUS_MODULE_NAME, MODELS_TO_MOVE))
```

**Review by sebalix — CHANGES_REQUESTED** [review]
The path of the migration script should be `migrations/14.0.1.0.0`, it's not a migration related to `14.0.1.1.0` specifically.

File: l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+QUERY_GET_NAMES = """
+    SELECT array_agg(name)
+    FROM ir_model_data
+    WHERE module = %s
+    AND model in %s;
+"""
+
+FIELD_NAMES_TO_MOVE = {
+    "product.product": [
+        "limited_quantity",
```
Comment (sebalix) [inline l10n_eu_product_adr/migrations/14.0.1.1.0/pre-dangerous-goods.py]: This field doesn't exist on 13.0, to remove, it makes the script crashing at some point.
```suggestion
```

**Review by TDu — APPROVED** [review]

**Review by sebalix — COMMENTED** [review]
A warning for the 13.0 users in the addon description is welcome to tell that dangerous goods attributes have been migrated to `l10n_eu_product_adr_dangerous_goods` and that they should install this addon themself to get their data back.

**Comment by sebalix** [comment]
I can not edit the description, but this PR aims to improve the state of https://github.com/OCA/community-data-files/pull/109.

**Pull request closed without merging**